### ISSUE TITLE
feat: Human Annotation & Compliance Review Queue (Low Priority)

### ISSUE THREAD

**Issue by brucearctor** [candelahq/candela#163]
## Summary

> **Low priority.** Human annotation only matters if the annotations feed into something — evals, fine-tuning datasets, or compliance audits. Standalone annotation ("let me label traces") without a downstream consumer is busywork. This feature should be built **after** the eval framework (#162) and enrichment SDK (#161) create consumers for annotation data.

The compliance angle is what makes this worth doing *eventually*: in healthcare/finance, you need audit trails that include human review. "A compliance officer reviewed and approved this LLM interaction" is a real regulatory requirement. A simple review queue in the dashboard — not a full annotation platform — serves this perfectly.

---

## Scope: What We're Building (and Not Building)

| Approach | Verdict |
|:---|:---|
| Full annotation queue system (assignments, workflows, inter-annotator agreement scoring) | ❌ Overkill — we're not building a labeling platform |
| **Simple thumbs-up/down on traces in the dashboard** | 🟢 Lightweight, immediately useful for identifying bad outputs |
| **Flag-for-review workflow** (mark trace → appears in compliance review queue) | 🟢 Fits the compliance narrative perfectly |
| **Export annotated traces to eval tools** | 🟢 Ties into the "integrate with eval tools" strategy (#162) |

This is a **thin layer on top of the trace store**, not a standalone product.

---

## Feature 1: Thumbs Up / Thumbs Down on Traces

The simplest possible annotation — a binary quality signal on any trace, directly in the dashboard.

### UX

- Every trace detail view gets a 👍 / 👎 button
- Optional: free-text comment field ("hallucinated the date", "wrong tone")
- Optional: category tag dropdown (e.g., `hallucination`, `wrong_format`, `good_response`, `off_topic`)
- Reviewer identity is captured automatically (logged-in dashboard user)

### Data Model

```json
{
  "trace_id": "abc-123",
  "annotation": {
    "rating": "thumbs_down",
    "comment": "Hallucinated a patient name that wasn't in the context",
    "category": "hallucination",
    "reviewer": "compliance-officer@acme.com",
    "timestamp": "2026-05-11T21:00:00Z"
  }
}
```

### API

```
POST /api/v1/traces/{trace_id}/annotate
{
  "rating": "thumbs_up" | "thumbs_down",
  "comment": "optional free text",
  "category": "optional category tag"
}
```

This is the same endpoint proposed in #161 (Enrichment SDK) — the SDK can submit annotations programmatically, and the dashboard uses the same API for manual annotations.

### Implementation

- [ ] Annotation data model (rating, comment, category, reviewer, timestamp)
- [ ] `POST /api/v1/traces/{trace_id}/annotate` endpoint
- [ ] Dashboard: thumbs up/down buttons on trace detail view
- [ ] Dashboard: optional comment and category fields
- [ ] Dashboard: annotation indicator on trace list view (annotated vs. not)
- [ ] Dashboard: filter traces by annotation status / rating

---

## Feature 2: Flag-for-Review Workflow

A lightweight compliance review queue — not a task management system.

### How It Works

```mermaid
graph LR
    A["Trace in Dashboard"] -->|"flag for review"| B["Review Queue"]
    B -->|"reviewer opens trace"| C["Trace Detail + Context"]
    C -->|"approve / reject + comment"| D["Reviewed Status"]
    D -->|"audit log entry"| E["Compliance Audit Trail"]
```

### UX

- Any trace can be flagged for review (manually, or automatically via rules)
- Dashboard gets a "Review Queue" view — list of flagged traces awaiting review
- Reviewer opens the trace, sees full context (prompt, response, cost, model)
- Reviewer marks as `approved` or `rejected` with optional comment
- Review decision is logged to the compliance audit trail

### Auto-Flag Rules

Traces can be automatically flagged based on configurable criteria:

```yaml
# candela config.yaml
review:
  auto_flag:
    - condition: "pii_detected == true"    # from guardrails (#159)
      reason: "PII was detected in this trace"
    - condition: "cost > 1.00"             # expensive requests
      reason: "High-cost request requires review"
    - condition: "error_rate > 0"           # failed requests
      reason: "Request resulted in an error"
    - condition: "model == 'gpt-4o'"        # specific model usage
      reason: "GPT-4o usage requires approval"
```

### Data Model

```json
{
  "trace_id": "abc-123",
  "review": {
    "status": "approved",
    "flagged_by": "auto:pii_detected",
    "flagged_at": "2026-05-11T21:00:00Z",
    "reviewed_by": "compliance-officer@acme.com",
    "reviewed_at": "2026-05-11T21:15:00Z",
    "decision": "approved",
    "comment": "PII was correctly redacted before reaching the LLM. Approved."
  }
}
```

### Implementation

- [ ] Flag-for-review action on traces (manual)
- [ ] Auto-flag rules engine (configurable conditions)
- [ ] Review queue dashboard view — list of flagged traces, sortable by date/reason
- [ ] Review action — approve / reject with comment
- [ ] Review status on trace detail view
- [ ] Review decisions logged to compliance audit trail (#159)
- [ ] Dashboard: filter by review status (pending, approved, rejected)

---

## Feature 3: Export Annotated Traces

Annotations are only valuable if they feed downstream consumers. Export targets:

| Consumer | How | Why |
|:---|:---|:---|
| **Eval tools** (promptfoo, deepeval) | Export via trace export API (#162) with annotations included | Annotated traces become ground truth for eval scoring |
| **Fine-tuning datasets** | Export approved 👍 traces as training examples (JSONL) | Human-vetted good responses → fine-tuning data |
| **Compliance reports** | Export review queue decisions as audit report (CSV/PDF) | "These N traces were reviewed and approved by X on date Y" |
| **Shadow eval scoring** | Annotations feed into shadow eval quality baselines (#162) | "Human-rated traces calibrate what 'good' looks like" |

### Implementation

- [ ] Include annotations in trace export API (#162)
- [ ] Export filter: only annotated traces, only 👍, only 👎, only reviewed
- [ ] Fine-tuning export format (JSONL with system/user/assistant turns from approved traces)
- [ ] Compliance report export (summary of review decisions, CSV)

---

## What We're Explicitly NOT Building

| Feature | Why Not |
|:---|:---|
| Task assignment system | We're not building a labeling platform — any reviewer can pick up any flagged trace |
| Inter-annotator agreement (IAA) | Overkill for compliance review — one reviewer per trace is sufficient |
| Annotation guidelines / instructions | Out of scope — compliance teams have their own review criteria |
| Batch annotation workflows | Low value — traces are reviewed individually in context |
| Paid annotator management | Not our business |

---

## Dependencies

| Dependency | Issue | Status |
|:---|:---|:---|
| Enrichment SDK (annotation API endpoint) | #161 | Shares the same `POST /traces/{trace_id}/annotate` endpoint |
| PII guardrails (auto-flag on PII detection) | #159 | Auto-flag rules trigger on guardrail events |
| Eval framework (export annotations as ground truth) | #162 | Annotations feed into eval scoring and shadow eval calibration |

---

## Priority

**Low priority.** This feature has limited value until its downstream consumers exist:
- Eval framework (#162) needs to be in place to consume annotations as ground truth
- PII guardrails (#159) need to be in place for auto-flag rules to trigger
- The enrichment SDK (#161) defines the annotation API endpoint

Build the consumers first. Then build the annotation layer that feeds them.

---

## Labels

`feature`, `compliance`, `low-priority`
